compiler/rustc_const_eval/src/check_consts/check.rs

```diff
@@ -172,7 +172,7 @@ pub struct Checker<'mir, 'tcx> {
 
     /// A set that stores for each local whether it is "transient", i.e. guaranteed to be dead
     /// when this MIR body returns.
-    transient_locals: Option<BitSet<Local>>,
+    transient_locals: Option<DenseBitSet<Local>>,
```

Comment:
Now that makes me wonder whether this is the right bit set type for this case? https://github.com/rust-lang/rust/pull/134438 switched a very similar case (also ranging over all locals in a MIR body) over to `MixedBitSet`.

But changing that should be a separate PR.

Reply:
@RalfJung Just checking: is this code path hit on the happy path or only when there are somewhat erroneous consts? It _feels_ like the latter. 

If so, it's possible that introducing a constant that trips up this check in a function with big locals could suddenly make compilation a slower, and I can open such a PR if you'd like to prevent this possible problem? It's good that the issue you point out shouldn't be as impactful as the `FlowSensitiveAnalysis` state from the other PR though.

Reply:
This will be computed for almost every const item that contains an `&` expression. So it definitely occurs on the happy path.

Reply:
I see, then we have no benchmarks that exercise this code path.

Reply:
Makes sense, thanks for looking into this!